OneCommunityGlobal/HGNRest
# Diya blue squares changes

# Description
Revoking the Edit Blue Squares permission ensures that the ‘Update’ button is not visible. Similarly, with the Delete Blue Squares and Assign Blue Squares permissions, the ‘Delete’ and ‘+’ buttons must not be visible to the user.
Update and Delete functionalities must update and delete the particular blue square.

## Related PRS (if any):
This backend PR is related to the backend PR #943.
To test this, you need to checkout PR #[2248](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2248) frontend PR

## Main changes explained:
- Commented out the check for the existing cache for a user so that the newly updated data in the database is fetched and displayed in realtime everytime the /userprofile API is called. 
- Added code to populate the infringements field when a user profile is being fetched.

## How to test:
1. check into current branch for the backend and the branch mentioned in the related frontend PR
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Log in as admin user
5. Other Links> Permissions Management>Volunteer>Assign/Edit/Delete Blue Squares
6. Log in as a volunteer and test all functionalities.

## Screenshots or videos of changes:
<img width="511" alt="Screenshot 2024-07-03 at 9 58 40 AM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/54880903/a38581ce-3e0d-4690-a7bc-5d8e2feeab0d">

## Note:
Please ensure you test the assign/edit/delete individually as well to ensure that each of the permissions has its own buttons visible as well as its own functionality.
